src/main/java/com/sap/sgs/phosphor/fosstars/model/feature/oss/OssFeatures.java

```diff
@@ -216,4 +217,53 @@ private OssFeatures() {
    */
   public static final Feature<Boolean> SIGNS_ARTIFACTS
       = new BooleanFeature("If a project signs artifacts");
+
+  /**
+   * If OWASP Dependency plugin is used always to scan during the build.
+   */
+  public static final Feature<Boolean> USES_OWASP_DEPENDENCY_CHECK_AS_MANDATORY_SCAN =
+      new BooleanFeature("If a project uses OWASP Dependency Check scan always");
+
+  /**
+   * If OWASP Dependency plugin is used always to report vulnerabilities during the build.
+   */
+  public static final Feature<Boolean> USES_OWASP_DEPENDENCY_CHECK_AS_MANDATORY_REPORT =
+      new BooleanFeature("If a project uses OWASP Dependency Check to generate report always");
+
+  /**
+   * If OWASP Dependency plugin is used maybe during the build.
+   */
+  public static final Feature<Boolean> USES_OWASP_DEPENDENCY_CHECK_AS_OPTIONAL_SCAN =
+      new BooleanFeature("If a project uses OWASP Dependency Check scan on specific occasions");
+
+  /**
+   * If OWASP Dependency plugin is used maybe to report vulnerabilities during the build.
+   */
+  public static final Feature<Boolean> USES_OWASP_DEPENDENCY_CHECK_AS_OPTIONAL_REPORT =
+      new BooleanFeature(
+          "If a project uses OWASP Dependency Check to generate report on specific occasions");
+
+  /**
+   * Check if configuration to fail the build exists in OWASP Dependency. Identify the CVSS
+   * threshold score to fail the build by failing junit.
+   */
+  public static final Feature<Double> OWASP_DEPENDENCY_CHECK_FAIL_JUNIT_CVSS_THRESHOLD =
+      new DoubleFeature(
+          "OWASP Dependency check CVSS Threshold score used to fail the junit tests");
+
+  /**
+   * Check if configuration to fail the build exists in OWASP Dependency. Identify the CVSS
+   * threshold score to fail the build on finding vulnerabilities greater than the threshold.
+   */
+  public static final Feature<Double> OWASP_DEPENDENCY_CHECK_FAIL_CVSS_THRESHOLD =
+      new DoubleFeature(
+          "OWASP Dependency check CVSS Threshold score used to fail the build");
+
+  /**
+   * Check if configuration to fail the build exists in OWASP Dependency. Identify whether the build
+   * fails if atleast one vulnerability is found.
+   */
+  public static final Feature<Boolean> USES_OWASP_DEPENDENCY_CHECK_FAIL_ANY_VULNERABILITY =
+      new BooleanFeature(
+          "If a project uses OWASP Dependency check to fail the build on finding vulnerability");
```

Comment:
Do we really need a separate feature for each parameter? In fact, is it allowed to use them together? Aren't they alternatives? I think they can be replaced by one double feature with a range [0, 11].

Reply:
Actually this is a good idea, if the 
- junitFailOnCVSS - then CVSS score is already provided
- failBuildOnCVSS - then CVSS score is already provided
- failBuildOnAnyVulnerability - then CVSS score is 0, as it considers all the vulnerabilities.

Reply:
> Actually this is a good idea, if the
> 
> * junitFailOnCVSS - then CVSS score is already provided
> * failBuildOnCVSS - then CVSS score is already provided
> * failBuildOnAnyVulnerability - then CVSS score is 0, as it considers all the vulnerabilities.

Yes, sounds good to me. And if none of the parameters are provided, then CVSS is set to 11 which means that the tool doesn't fail if any issue is found. It looks a bit ugly though. Another way is to add a boolean feature that is set to true if no threshold provided.